config/base.py

```diff
@@ -0,0 +1,144 @@
+"""Base configuration holder."""
+
+from __future__ import annotations
+from typing import TYPE_CHECKING, List, Optional, TypeVar, Callable, Union
+
+if TYPE_CHECKING:
+    # Avoid cyclic imports used for typing only.
+    from database.tables import ConfigsTable
+
+
+T = TypeVar("T")
+
+
+class ConfigVariable():
```

Comment:
nit: unnecessary parenthesis
```suggestion
class ConfigVariable:
```